component/main.jsonnet

```
@@ -69,16 +49,29 @@ local identityProviders = [
 
   idp {
     ldap+: {
-      ca: { name: authentication.RefName(idp.name) },
-      bindPassword: { name: authentication.RefName(idp.name) },
+      ca: { name: common.RefName(idp.name) },
+      bindPassword: {
+        name: if std.objectHas(idp, 'bindPasswordSecretRef') then params.secrets[idp.bindPasswordSecretRef] else common.RefName(idp.name),
+      },
```

Alternative approach to make this backwards compatible without introducing an extra field which might break OCP:

```suggestion
      bindPassword:
        if std.isString(super.bindPassword) then
          // Legacy variant: value of `bindPassword` is a string, so we inject the secret name for the generated legacy secret
          { name: common.RefName(idp.name) }
        else
          // In other cases: Just reuse the original value for `bindPassword`, leave it to the user to correctly format it.
          super.bindPassword,
```